packages/expo-file-system/src/next/FileSystem.types.ts

```diff
@@ -29,17 +29,23 @@ export declare class Directory {
   validatePath(): void;
 
   /**
-   * Deletes a directory.
+   * Deletes a directory. Also deletes all files and directories inside the directory.
+   *
+   * @throws `UnableToDeleteException` If the directory does not exist or cannot be deleted.
```

Comment:
To be honest I'm not sure if we're ready to document thrown errors. How do people check if the error is of `UnableToDeleteException` type or something else? Right now the proper way would be to check the `code` of the error.